lib/solarwinds_apm/otel_native_config.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+# © 2023 SolarWinds Worldwide, LLC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at:http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+require 'solarwinds_apm/constants'
+require 'solarwinds_apm/api'
+require 'solarwinds_apm/support'
+require 'solarwinds_apm/opentelemetry'
+require 'solarwinds_apm/sampling'
+
+module SolarWindsAPM
+  # OTelNativeConfig module
+  module OTelNativeConfig
+    @@config           = {}
+    @@config_map       = {}
+    @@agent_enabled    = true
+
+    def self.initialize
+      return unless defined?(::OpenTelemetry::SDK::Configurator)
+
+      ENV['OTEL_TRACES_EXPORTER'] = ENV['OTEL_TRACES_EXPORTER'].to_s.split(',').tap { |e| e << 'otlp' unless e.include?('otlp') }.join(',')
+      ENV['OTEL_RESOURCE_ATTRIBUTES'] = "sw.apm.version=#{SolarWindsAPM::Version::STRING},sw.data.module=apm,service.name=#{ENV.fetch('OTEL_SERVICE_NAME', nil)}," + ENV['OTEL_RESOURCE_ATTRIBUTES'].to_s
```

Comment:
Is `OTEL_SERVICE_NAME` always set? If it's not, then does service name come from `SW_APM_SERVICE_KEY`?

Reply:
Sorry for the confusion. I have another PR that configure service name; that will make sure `OTEL_SERVICE_NAME` will be set either derived from `SW_APM_SERVICE_KEY` or not